### ISSUE TITLE
TB 78 : Quick text "disabled" often

### ISSUE THREAD

**Issue by PotatoCarl** [thundernest/quicktext#167]
Hi
Quicktext is "enabled" under add ons, but not show (and keyboard shortcuts not working) in a new message/reply to.
When I go under "add ons" and disable and reenable quick text, it is displayed.
In the next email, same is happening.
Gruss
TB

**Comment by stb2015** [comment]
I have the same problem.

**Comment by abernitz** [comment]
Hello,

same problem here

Regards
Alex

**Comment by jobisoft** [comment]
Can you try 3.2?
[quicktext_3.2.xpi](https://github.com/jobisoft/quicktext/files/5102434/quicktext.zip)

**Comment by jobisoft** [comment]
Quicktext 3.2 officially released.

**Comment by PotatoCarl** [comment]
Same problem. When I startup TB it is shown in the addons "active" but no menu or functionality. Have to turn it off and on to make it work.

**Comment by jobisoft** [comment]
QT3.2? What other add-ons are you using?

**Comment by PotatoCarl** [comment]
Yes. TBSync, Category Manager, Compact Header, Encrypt if Possible, Colored Folders

**Comment by jobisoft** [comment]
Category Manager does not work in TB78. Are you on TB78?

**Comment by PotatoCarl** [comment]
It seems to work fine. Yes. I am on TB78.

**Comment by jobisoft** [comment]
I am the author of CatMan. I did not release a version for TB78. The latest version 3.11 can by no means work in TB78. Do not know what is going on. I assume you are on TB68. But than the version 3.2 of Quicktext is wrong. *puzzled*

**Comment by PotatoCarl** [comment]
CategoryManager Version 3.11
It is marked as "active" after update to TB78. It claims to be working. But after opening the Addon a text is displayed, that it is not working and should not be installed. Confusing.
I disabled it.
QT is 3.2 according to Addonpage.

**Comment by PotatoCarl** [comment]
CategoryManager is also not listed as "incompatible" in the Addon overview of TB78...

**Comment by jobisoft** [comment]
On ATN it is clearly listed as not compatible and when you search for CatMan in the Thunderbird AddonManger you should not get it installed. Only by manual file installation, correct? That is a bug we are working on (in Thunderbird).

This is probably the same for:
- Encrypt if Possible
- Colored Folders
- Compact Header

Can you confirm? So the only other addon compatible with TB78 is TbSync, right? Which version are you using?

**Comment by PotatoCarl** [comment]
I upgraded to TB78 by installing TB78 over TB68. addons were partly marked "incompatible and deactivated", but not CM.
It was not installed by file.
Encrypt if possible and compact header both say working and fine, but do in fact not work, you are right (did not notice yet).
Colored Folders seems to work, as the folders can in fact be colored nicely...
For TBSync the activesync provider of course also.
TBsync is 2.16., Provider 1.16
I deactived not everything except TBsync, Provider, QT and ColoredFolders.

**Comment by PotatoCarl** [comment]
oh, an installed the Pre-Version of Quickfolders just right now ;-)

**Comment by PotatoCarl** [comment]
now it works - when I disabled the other addons

**Comment by jobisoft** [comment]
Thanks for that report. We need to fix that in core, that addons which do not actually work in TB78 must disabled.

So this boils down to an incompatibiliyt between Quicktext and Quickfolders? If you disable Quickfolders, the issue does not happen, but if you enable Quickfolders, than Quicktext reproducable disables itself from time to time?

Thanks for your patience.

**Comment by PotatoCarl** [comment]
Incompatible with either Compact header, category manager or encrypt if possible.
With Quickfolders no problem.

**Comment by jobisoft** [comment]
A landed a patch wich disables unsupported add-ons:
https://bugzilla.mozilla.org/show_bug.cgi?id=1661216

Should be in the next release of TB 78.2.2

**Comment by Thomas-1985** [comment]
Incompatible with EnigMail (2.2.2) in TB 78.2.1. If Enigmail is enabled (it is marked as a supported addon) the Quicktext menu bar  when compiling an email is not shown anymore.

**Comment by thomiel** [comment]
As a workaround: Enigmail is only used for migrating GPG keys. After you have imported all you keyrings, you can safely disable enigmail and rely on the OpenPGP implementaion that is integrated in TB 78.

**Comment by EmmyGraugans** [comment]
Sadly, I have the same problem, even without *any* other enabled AddOns.
Thunderbird 78.2.2 with German Localization under macOS, if that makes any difference. I haven't yet tried under another operating system (still in Thunderbird 68.x there)

I used to have installed Enigmail and Mail Redirect, Enigmail I removed after migration of GPG, Mail Redirect was disabled, as it is yet incompatible with the new Thunderbird - as a test, I removed the disabled AddOn now, but it makes no difference.

Disabling and re-enabling Quicktext makes it appear in the "Compose" window and the "Extras" Menu for Configuration, but after a restart of Thunderbird, both vanish again.

As an additional test, I made a new, *empty* profile (even without registering a new Email-Address), and there Thunderbird shows the Quicktext-Configuration menu without having to disable and re-enable. So there must be some remnants in the profile by migrating from the older version that inhibit Quicktext in the new version.

**Comment by SamuelPlentz** [comment]
You could probably find the bad file by duplicating your bad profile and delete one file / folder at a time and restart Thunderbird. Thunderbird will usually create blank files instead on startup. If it works the deleted file might be the cause. Create another duplicate, only delete that file and restart Thunderbird, to make sure.

Google that file. If it is a text file use some program like WinMerge to see what differences are there. Remove different lines until a restarted Thunderbird works. Good Luck.

**Comment by EmmyGraugans** [comment]
@SamuelPlentz : not a bad idea. Deleting the files "key4.db" and "secring.gpg" made Quicktext work reliably... Of course, it deleted my passwords and (most of) the PGP-Configuration... The really interesting bit is now, I re-entered my passwords (not yet PGP-Configuration) and it still worked on restart of Thunderbird. I re-set a master-password and Quicktext stopped working again! I deleted the master-password and it started working again.

Now I went back to my original old profile and deleted the master-password and - Quicktext works! (Without deleting any files) Just, setting a master-password disables it again on restart of Thunderbird until I manually disable and re-enable the AddOn.

Last test: I took the clean Thunderbird-Profile again with only installed Quicktext-AddOn, set a master-password there - and it stopped working. Conclusion: Quicktext doesn't like a set master-password. Whyever...

**Comment by EmmyGraugans** [comment]
Addition, not sure if relevant or not: Since the Upgrade to Thunderbird 78.x, I have to enter my master-password *twice*. That happened also on the clean profile without yet any email-addresses.

**Comment by jobisoft** [comment]
> Conclusion: Quicktext doesn't like a set master-password. Whyever...

I will have a look. Thanks for your research.

**Comment by thomiel** [comment]
I don't have any trouble with a set master password...

**Comment by jobisoft** [comment]
As the OS may be related, can you all include your OS in any further works yes/no comment? That would be helpful. Thanks!

**Comment by thomiel** [comment]
Windows 10 Version 2004 build 19041.450 (works with disabled Enigmail, doesn't work with enabled Enigmail)

**Comment by joostn** [comment]
Same problem here, with a master password set, QuickText does not work (as if it were not installed at all).

Thunderbird 78.2.2 (64-bit), macOS 10.15.6

**Comment by jobisoft** [comment]
> 
> 
> Windows 10 Version 2004 build 19041.450 (works with disabled Enigmail, doesn't work with enabled Enigmail)

I have only Windows 10 Version 1909 and cannot reproduce that :-(

**Comment by olivier-15kay** [comment]
Same issue here. 

- **Problem went away when I disabled Enigmail and restarted TB.**
- Re-enabling Enigmail makes QuickText disappear again after TB restart. 
- Nothing obvious in the error console, but not an expert here. Any suggestion where to look?

Windows 10 pro 2004 19041.508, TB 78.2.2 (64-bit), QuickText 3.2

**Comment by BayouBillLa** [comment]
I am having this same problem with TB 78.2.2 (32-bit) and Quicktext 3.2, which shows a "last updated" date of September 15, 2020. I have been using Quicktext for years without a problem until now. I have no other add-ons installed. Disabling and then reenabling Quicktext restores the function, but then it disappears again after a while. I'm running Windows 10 Pro 64-bit 1909 18363.1082.

**Comment by jobisoft** [comment]
Please watch the error console (CTRL+SHIFT+J) for Quicktext related errors. I have no idea.

**Comment by SamuelPlentz** [comment]
@BayouBillLa Do you have a Masterpassword set? Does removing it help?
See [comment 25](https://github.com/jobisoft/quicktext/issues/167#issuecomment-691047879_) from above.

Does it work with an empty profile? You could try this:
See [comment 24](https://github.com/jobisoft/quicktext/issues/167#issuecomment-690972876_) from above.

**Comment by hagedose** [comment]
> @BayouBillLa Do you have a Masterpassword set? Does removing it help?
> See [comment 25](https://github.com/jobisoft/quicktext/issues/167#issuecomment-691047879_) from above.

I'm on a Mac with 10.15.6, and removing the master password helped. Not that that's a permanent solution ...

**Comment by ru320** [comment]
Hello everybody
I had the same problem.

**The problem was resolved after I:**

- Removed Enigmail (_The Enigmail functionality is now part of Thunderbird. Therefore problems were also to be expected_)
- All other ADDON also got removed.
- Cleared the cache.

Then I was able to import all ADDONS without any problems.

**Comment by PaulShave** [comment]
The discussion seems to have widened out since it was first reported. I still have the same problem of having to disable and re-enable Quicktext each time TB is reloaded. Running Windows 10 19041.508 with TB 78.3.1 and Quicktext 3.2. The only other addons I have are ImportExportTools NG and Manually sort folders.
I don't know if it's relevant, but there's an error console message:
Promise rejected after context unloaded: Actor 'Conduits' destroyed before query 'RuntimeMessage' was resolved
5 content_script_bundle.js:110
    action moz-extension://ace1be88-c474-49c9-b47e-b8571249be02/dist/content_script_bundle.js:110

**Comment by jobisoft** [comment]
Can you disable manual sort folders? I believe it is not compatible with TB78.

**Comment by PaulShave** [comment]
Manual sort folders is Version 2.0.4, last updated 28/09/2020. But I've disabled it and reloaded TB, and the Signature bar now appears, so hopefully that was the problem.

**Comment by jobisoft** [comment]
I checked the source of MSF 2.0.4 and it looks like it **is** compatible. If you install MSF anew, QT fails again?

**Comment by PaulShave** [comment]
I've only ever used MSF twice, so I'm happy to leave it disabled and only enable it if and when I want to use it.

**Comment by daCaPo** [comment]
> I checked the source of MSF 2.0.4 and it looks like it **is** compatible. If you install MSF anew, QT fails again?

I've MSF 2.0.4. installed and activated, and currently Quick text loads fine at startup.

**Comment by SamuelPlentz** [comment]
For your information: I tried to reproduce this bug in TB 78.3.1 (Win 10) with Masterpassword set or some of the Addons in use, but couldn't. Is this bug still present in TB 78.3.1 for anyone?

The only thought I have is that the TB error console (CTRL+SHIFT+J) has a Quicktext related error, but it slipped through because it didn't seem to be related. Probably the first ones of the following files are not so much connected to quicktext:
- background.js
- implementation.js
- main.js
- messenger.js
- messengercompose.js
- preferences.js
- settings.js
- startup.js
- utils.jsm
- quicktext.js
- wzQuicktext.jsm
- wzQuicktextGroup.jsm
- wzQuicktextHeader.jsm
- wzQuicktextScript.jsm
- wzQuicktextTemplate.jsm
- wzQuicktextVar.jsm

Can anyone that can still reproduce check if there are entries in the TB error console for one of this files?

**Comment by EmmyGraugans** [comment]
Yes, I can reproduce the error still in MacOS with Thunderbird 78.3.1, but also yes, I can NOT reproduce the problem in Windows (1909 Home) with the same Thunderbird-version. I tried looking at the Error-Console, and it seems that QuickText isn't even beginning to load. I took again an empty profile, once with master-password and once without.

Without master-password, I get these lines somewhere in the error-log:
```
WebExtensions: Waiting for async startup() in <chrome://quicktext/content/scripts/startup.js> to finish. implementation.js:202
WebExtensions: startup() in <chrome://quicktext/content/scripts/startup.js> finished implementation.js:205
```

WITH a master-password, these lines appear only after disabling and re-enabling QuickText. All other lines are identical in both variants. I am not sure if that really means that Thunderbird isn't even trying to start QuickText (or *any* WebExtension? Does somebody know another "WebExtension" that one could try, that's still compatible with Thunderbird 78?)

Maybe important, but probably not - my Thunderbird is running in German Localization

I was curious enough to test Thunderbird 78.3.1 also under Linux - the problem does not occur there. With empty profile and activated master-password, QuickText is still available right from the start... The problem is apparently somehow macOS-specific...

**Comment by SamuelPlentz** [comment]
Sounds like the master-password form prevents / delays and timeouts the execution of the startup script.

**Comment by joostn** [comment]
The problem is still there in macOS. I'm using Thunderbird in English. Under Windows all is fine. I'm seeing two prompts for the master password, the second prompt appears while the first is still shown on screen.

**Comment by EmmyGraugans** [comment]
> I'm seeing two prompts for the master password, the second prompt appears while the first is still shown on screen.

Same here - and that also happens only in macOS (that is NOT a QuickText Issue, but it might be related to the reason for the missing QuickText). In case that is in any way relevant: Quitting the first password prompt and answering the second password prompt works, answering the first password prompt and quitting the second does NOT work. Makes no difference to QuickText however, whether answering one, both, or none of the password-prompts.

**Comment by BayouBillLa** [comment]
I have no other add-ons installed, and Quicktext is still failing. Windows 10 pro 64-bit 1909 18363.1082, Thunderbird 78.3.1 (32-bit), Quicktext 3.2 last updated 9/15/20.

 I've attached three logs. 

Log1 is after a normal startup, after which Quicktext is not available when composing a new message.
Log2 is when Quicktext becomes available again after disabling and reenabling it.
Log3 is when Quicktext is not available when restarting Thunderbird immediately after disabling then reenabling Quicktext.

Note that there are no references to Quicktext in Log1 or Log3, which may mean that it is not getting loaded at all at startup. Log2 clearly shows it being loaded. 

Hope this helps.
[Log1NormalStartup.txt](https://github.com/jobisoft/quicktext/files/5341579/Log1NormalStartup.txt)
[Log2AfterDisablingAndReenabling.txt](https://github.com/jobisoft/quicktext/files/5341583/Log2AfterDisablingAndReenabling.txt)
[Log3QuicktextFailureAfterRestart.txt](https://github.com/jobisoft/quicktext/files/5341584/Log3QuicktextFailureAfterRestart.txt)

**Comment by SamuelPlentz** [comment]
Could you please try what this version writes to the error console (CTRL + ALT + J)?

- [quicktext-3.3a2-tb.xpi.zip](https://github.com/jobisoft/quicktext/files/5343038/quicktext-3.3a2-tb.xpi.zip) (remove the ".zip" extension and install it)

It should create log entries like this, to let us know, where the processing stops:
```
background.js - detailed log #1
background.js - detailed log #2
...
background.js - detailed log #10
WebExtensions: implementation.js - detailed log #1
WebExtensions: implementation.js - detailed log #2
...
WebExtensions: implementation.js - detailed log #5
```

**Comment by BayouBillLa** [comment]
The attached log is after installing 3.3.a2, then clearing the log, restarting Thunderbird, and starting a new email. The Quicktext toolbar was active in the composition window of that email.

I will keep testing and post another log file if Quicktext fails again.

[LogB1_AfterInstalling3.3.a2AndStartingANewEmail.txt](https://github.com/jobisoft/quicktext/files/5343288/LogB1_AfterInstalling3.3.a2AndStartingANewEmail.txt)

**Comment by SamuelPlentz** [comment]
Strange I didn't change anything from 3.2 but to add some debug messages. The case if it doesn't load as intended is interesting.

**Comment by hagedose** [comment]
I'm on macOS, where Quicktext only works when no master passphrase is set. I installed 3.3.a2, enabled the master passphrase, and the following is logged by background.js:

background.js - detailed log #1 background.js:3:11
background.js - detailed log #2 background.js:19:11

I don't see anything from WebExtensions: implementation.js

**Comment by SamuelPlentz** [comment]
@hagedose Thank you that is what we need. Just to make sure: There are background.js entries from `#1` to `#10`. They end at `#2`, correct?

**Comment by hagedose** [comment]
Exactly.

**Comment by jobisoft** [comment]
It looks like the master password prevents the add-on from accessing its storage, but it does not simply delay the execution of the add-on, it stops it, as the "await" after log #2 never fulfills. That is a stupid bug in TB. I will investigate. For the time being, there is no way to check in background.js, if the password has been entered already. Not that I know of at least, checking.

@SamuelPlentz : You do not need to invest more time to test this. I will try to come up with a solution.

**Comment by SamuelPlentz** [comment]
@jobisoft Nice to hear that. Good luck!

As I also created this next version for detailled "preferences.js" messages I will post it:

@hagedose / others: Could you please try what this version writes to the error console (CTRL + ALT + J)?

- [quicktext-3.3a4-tb.xpi.zip](https://github.com/jobisoft/quicktext/files/5343567/quicktext-3.3a4-tb.xpi.zip) (remove the ".zip" extension and install it)

It should create log entries like this, to let us know, where the processing stops:
```
background.js - detailed log #1
background.js - detailed log #2
...
background.js - detailed log #10
WebExtensions: implementation.js - detailed log #1
WebExtensions: implementation.js - detailed log #2
...
WebExtensions: implementation.js - detailed log #5
preferences.js - detailed log #1
preferences.js - detailed log #2
...
preferences.js - detailed log #12
```

**Comment by jobisoft** [comment]
Ok, that is now a test version from me, based on the current master incl. the waitForMasterPassword. No need to rename that zip to xpi, you can install it as is. What do you see on the console?

Edit: File removed.

**Comment by jobisoft** [comment]
One of @SamuelPlentz patches went boom. Hold on.

**Comment by hagedose** [comment]
> @jobisoft Nice to hear that. Good luck!
> 
> As I also created this next version for detailled "preferences.js" messages I will post it:
> 
> @hagedose / others: Could you please try what this version writes to the error console (CTRL + ALT + J)?
> 
> * [quicktext-3.3a4-tb.xpi.zip](https://github.com/jobisoft/quicktext/files/5343567/quicktext-3.3a4-tb.xpi.zip) (remove the ".zip" extension and install it)
> 
> It should create log entries like this, to let us know, where the processing stops:
> 
> ```
> background.js - detailed log #1
> background.js - detailed log #2
> ...
> background.js - detailed log #10
> WebExtensions: implementation.js - detailed log #1
> WebExtensions: implementation.js - detailed log #2
> ...
> WebExtensions: implementation.js - detailed log #5
> preferences.js - detailed log #1
> preferences.js - detailed log #2
> ...
> preferences.js - detailed log #12
> ```

This is the last entry:

preferences.js - detailed log #5

**Comment by hagedose** [comment]
> [quicktext.zip](https://github.com/jobisoft/quicktext/files/5343611/quicktext.zip)
> 
> Ok, that is now a test version from me, based on the current master incl. the waitForMasterPassword. No need to rename that zip to xpi, you can install it as is. What do you see on the console?

WebExtensions: Waiting for master password. implementation.js:114
this.sleep is not a function implementation.js:115
    waitForMasterPassword jar:file:///Users/hgd/Library/Thunderbird/Profiles/45obt18g.default/extensions/{8845E3B3-E8FB-40E2-95E9-EC40294818C4}.xpi!/api/WindowListener/implementation.js:115
    waitForMasterPassword self-hosted:844
    result resource://gre/modules/ExtensionParent.jsm:996
    withPendingBrowser resource://gre/modules/ExtensionParent.jsm:602
    result resource://gre/modules/ExtensionParent.jsm:996
    callAndLog resource://gre/modules/ExtensionParent.jsm:958
    recvAPICall resource://gre/modules/ExtensionParent.jsm:995
    AsyncFunctionNext self-hosted:693
    openWindowPrompt resource://gre/modules/Prompter.jsm:1229
    openPrompt resource://gre/modules/Prompter.jsm:1087
    openPromptSync resource://gre/modules/Prompter.jsm:1070
    nsIPrompt_promptPassword resource://gre/modules/Prompter.jsm:1559
    promptPassword resource://gre/modules/Prompter.jsm:1291
    retrieveOpenPGPPassword chrome://openpgp/content/modules/masterpass.jsm:109
    haveMasterPassword chrome://openpgp/content/modules/masterpass.jsm:56
    init chrome://openpgp/content/BondOpenPGP.jsm:102
    InterpretGeneratorResume self-hosted:1151
    AsyncFunctionNext self-hosted:693

**Comment by SamuelPlentz** [comment]
``` Javascript
console.log("preferences.js - detailed log #5");

// Remove all defaults from storage.local for which we do not have an
// entry in the provided defaults obj.
let defaultPrefStorage = await messenger.storage.local.get();

console.log("preferences.js - detailed log #6");
```

**Comment by jobisoft** [comment]
@SamuelPlentz : Yep, that is the storage access failing.

I fixed the bugs I had in my own test version and it worked for me:
[quicktext.zip](https://github.com/jobisoft/quicktext/files/5343718/quicktext.zip)

No need to rename that to xpi, you can install it as it is.

What happens?

**Comment by hagedose** [comment]
FWIW, there was no need to rename any of the add-ons. With the new one I see

`WebExtensions: Waiting for master password. implementation.js:127
`
several times, and then:

`WebExtensions: Master password has been entered. implementation.js:130
`
Alas, Quicktext still doesn't work until I disable and re-enable it.

**Comment by jobisoft** [comment]
How many password prompts do you get? One or two? If more than one, could you dismiss the first one?

And even though QT is waiting for the password prompt, it still fails to access its storage? What a mess...

**Comment by hagedose** [comment]
There are two password prompts. If I dismiss the first one (twice), and enter the password in the second one, it works.

**Comment by hagedose** [comment]
The add-on Copy Sent to Current appears to have the same issue. What should I tell the developer of that add-on to do?

**Comment by BayouBillLa** [comment]
The attached log is after installing 3.3.a4, then clearing the log, restarting Thunderbird, and starting a new email. The Quicktext toolbar was active in the composition window of that email.

[LogC1_AfterInstalling3.3.a4AndStartingANewEmail.txt](https://github.com/jobisoft/quicktext/files/5343827/LogC1_AfterInstalling3.3.a4AndStartingANewEmail.txt)

**Comment by jobisoft** [comment]
@hagedose : Not much, as we have not really fixed it and what you see is a bug in TB. Add-on startup in general should wait till after the user is logged id.

**Comment by jobisoft** [comment]
@SamuelPlentz : Could you take over again? You have the better overview of where which user is at. It appears as if the current master branch works, if the password prompt is dismissed a few times, before it gets entered.

Would be interesting to see, if this can be confirmed by all users.

**Comment by EmmyGraugans** [comment]
The version from jobisoft's recent post #issuecomment-705189201 works correctly, when I enter the password.

Systematically:
- entering the password first in the first password-form, then again in the second password-form
  --> works
- entering the password first in the first password-form, then quitting the second password-form 
  --> works
- waiting on the first password form, until the second appears, then entering the password in the second password-form, and afterwards quitting the first password-form
  --> DOES NOT WORK (Quicktext doesn't appear before being disables and re-enabled)
- waiting on the first password form, until the second appears, then entering the password in the second password-form, and afterwards entering the password in the first password-form as well
  --> DOES NOT WORK (Quicktext doesn't appear before being disables and re-enabled)

So it seems that for Quicktext to work, the first password-prompt must be served first

I am quite sure that in earlier tests in some of the orders of password-entry, Thunderbird didn't have a working PGP-Configuration, but that might have been fixed in the meantime. At least now the order of passwords (or quitting of the one or the other password-form) didn't seem to make a difference to email-access, or to PGP, only to the availability of Quicktext

**Comment by jh20001** [comment]
To chime in, I experienced the plugin's menus/options vanishing just once. Ver 3.2 of the plugin has been working fine in TB78 before that happened. I disabled and then re-enabled and everything came back and it hasn't happened again. I am pretty sure this has happened "once" in the past as well in an older version of TB. So this might be a blip in the Matrix for me.

**Comment by BayouBillLa** [comment]
After working well for 24 hours or so, Quicktext 3.3.a4 failed to load when I restarted TB just now. Same environment as I detailed in yesterday's post. Log is attached.

[LogD1_Quicktext3.3.a4FailsToLoadAfterRestartingTB.txt](https://github.com/jobisoft/quicktext/files/5351766/LogD1_Quicktext3.3.a4FailsToLoadAfterRestartingTB.txt)

**Comment by SamuelPlentz** [comment]
Here is an overview - I hope everything is correct:
![Disable](https://user-images.githubusercontent.com/1890854/95625592-010ecd80-0a79-11eb-8046-50e921d6452f.png)

I) We clearly have the problem in windows, that bad Addons prevent Quicktext from working. That issue should be solved as newer versions of Thunderbird automatically disable bad Addons. Last reported 2020-10-03 (7 days ago)

II) We have another problem in macOS, that a Masterpassword prevents Quicktext from working.

We don't know what the problem for @stb2015 and @abernitz is. They both only stated "same here", but we neither have a list of Addons in use, nor the OS or if a Masterpassword is set.

III) And then there is @BayouBillLa who has a problem on a windows machine without any Addons and without a Masterpassword set.

@BayouBillLa: Could you please confirm, that there are no Addons installed except Quicktext? Can you try to use a new and empty profile? Can you reproduce the problem there?

If you cannot reproduce: You could probably find the bad file of your current profile by duplicating your bad profile and delete one file / folder at a time and restart Thunderbird. Thunderbird will usually create blank files instead on startup. If it works the deleted file might be the cause. Create another duplicate, only delete that file and restart Thunderbird, to make sure.

Report back that file. If it is a text file use some program like WinMerge to see what differences are there. Remove different lines until a restarted Thunderbird works. Good Luck.

**Comment by SamuelPlentz** [comment]
Edit: [Text removed]

Edit: Probably won't change anything. The only reason the Addon chrashes is when accessing the storage. Many Addons don't do that.

**Comment by SamuelPlentz** [comment]
So the remaining puzzles to solve:

1. How could it be prevented, that accessing the storage disables the addon, if the macOS Masterpassword is not entered correctly or in the case of @BayouBillLa?
2. What is the reason that prevents Quicktext from working in @BayouBillLa environment? ([see the lower part of my post above](https://github.com/jobisoft/quicktext/issues/167#issuecomment-706378857))
3. Anyone that had problems with bad Addons and can still reproduce it (@Thomas-1985, @thomiel, @PotatoCarl, @olivier-15kay, @ru320, @PaulShave) could try if the version 3.3a4 also stops when accessing the storage. Last entry `preferences.js - detailed log #5` in error console (CTRL+SHIFT+J) when it gets automatically disabled.

[quicktext-3.3a4-tb.xpi.zip](https://github.com/jobisoft/quicktext/files/5343567/quicktext-3.3a4-tb.xpi.zip) (remove the ".zip" extension and install it)

Perhaps we should focus the code on is the storage available instead of is the Masterpassword entered...
Perhaps a try and catch around it in a while loop with sleep between each try.

**Comment by BayouBillLa** [comment]
I will try to do some more testing, as requested, over the weekend and 
will report back.

BayouBillLa

**Comment by SamuelPlentz** [comment]
Created an version that tries to catch errors and retries accessing the storage:
[quicktext-3.3a5-tb.zip](https://github.com/jobisoft/quicktext/files/5358687/quicktext-3.3a5-tb.zip)

Could you try if that works for you, if you have the disable problem? If it doesn't work, can you please attach the content of your thunderbird error log? (CTRL+SHIFT+J)

**Comment by jobisoft** [comment]
> Perhaps we should focus the code on is the storage available instead of is the Masterpassword entered...
Perhaps a try and catch around it in a while loop with sleep between each try.

That does not work, as the storage access does not fail. It just never returns and stops execution flow. You cannot detect that just by accessing the storage.

**Comment by SamuelPlentz** [comment]
Yep also thought that could be the case. (run forever instead of throwing an exception)
Wanted to exclude that possibility.

So what do you propose? Solve that in Thunderbird itself? (load Addons after Masterpasswords have been entered / after the storage is ready) Should we create a bug report?

If you are right, then the only other possibilities I can think of are:
1. Find some kind of flag or property of Thunderbird, that is only set after the initialization.
2. Create a thread trying to access the storage. If the thread doesn't return after some seconds: Kill it and try again.
3. Use a file for properties (like `templates.xml` and `scripts.xml`).

**Comment by jobisoft** [comment]
This version delays add-on startup until the main window has been defined and mail-startup-done.
[quicktext.zip](https://github.com/jobisoft/quicktext/files/5358792/quicktext.zip)


If the add-on is activated, while TB is running, you will see:
`WebExtensions: Startup seems to be done.`

After a fresh TB start, you will see something like this on the console:
`WebExtensions: Waiting for mail-startup-done.`

After mail-startup-done has been seen, he will check for the window state and master password. So you will probably see:
`WebExtensions: mail-startup-done has been seen.`
`WebExtensions: Main window state is: complete`
`WebExtensions: Master password has been entered.`

or - if the password has not been entered - 
`WebExtensions: Waiting for master password.`

I actually hope, that the Password stuff is done BEFORE mail-startup-done is triggered, so I hope you do not see the "waiting for master password" log entry.

Can you all try this version, regardless of your OS and report back stating which OS you use, if a master password is used and what you see on the console - and if QT works of course.

**Comment by hagedose** [comment]
With the new version QT does _not_ work on macOS with master password enabled. Console output:

WebExtensions: Startup seems to be done. implementation.js:414
WebExtensions: Waiting for master password. implementation.js:129
WebExtensions: Waiting for master password. implementation.js:129
WebExtensions: Waiting for master password. implementation.js:129
WebExtensions: Master password has been entered. implementation.js:132

I do not see any lines referring to mail-startup-done.

**Comment by jobisoft** [comment]
@hagedose : You really see that during a Thunderbird start with the QT version enabled?

**Comment by hagedose** [comment]
Yes. I can see the code in implementations.js.

**Comment by EmmyGraugans** [comment]
> @hagedose : You really see that during a Thunderbird start with the QT version enabled?

I can confirm that (on macOS). Your (jobisoft) Version from 5 days ago works, the version from 2 days ago doesn't. (Also confirmed by again "downdating")The version "3.3a5" by SamuelPlentz also works with the following log:

```
background.js - v2 detailed log #1a background.js:3:11
WebExtensions: Waiting for master password. implementation.js:127
WebExtensions: Waiting for master password. implementation.js:127
WebExtensions: Waiting for master password. implementation.js:127
WebExtensions: Waiting for master password. 6 implementation.js:127
background.js - v2 detailed log #1b background.js:7:11
WebExtensions: Master password has been entered. implementation.js:130
background.js - v2 detailed log #2 background.js:23:11
preferences.js - v2 detailed log #1 preferences.js:95:13
preferences.js - v2 detailed log #2 preferences.js:99:13
preferences.js - v2 detailed log #3 preferences.js:109:13
preferences.js - v2 detailed log #4 preferences.js:114:13
preferences.js - v2 detailed log #5a preferences.js:120:15
preferences.js - v2 detailed log #5b preferences.js:134:19
preferences.js - v2 detailed log #5c preferences.js:140:19
preferences.js - v2 detailed log #5d preferences.js:151:19
preferences.js - v2 detailed log #7 preferences.js:159:15
preferences.js - v2 detailed log #8 preferences.js:167:15
preferences.js - v2 detailed log #12 preferences.js:188:13
background.js - v2 detailed log #3 background.js:27:11
background.js - v2 detailed log #4 background.js:49:11
background.js - v2 detailed log #5 background.js:66:11
background.js - v2 detailed log #6 background.js:70:11
background.js - v2 detailed log #7 background.js:74:11
background.js - v2 detailed log #8 background.js:80:11
background.js - v2 detailed log #9 background.js:86:11
background.js - v2 detailed log #10 background.js:90:11
preferences.js - v2 detailed log #1 preferences.js:95:13
preferences.js - v2 detailed log #2 preferences.js:99:13
preferences.js - v2 detailed log #3 preferences.js:109:13
preferences.js - v2 detailed log #4 preferences.js:114:13
preferences.js - v2 detailed log #9 preferences.js:171:15
preferences.js - v2 detailed log #10 preferences.js:176:15
preferences.js - v2 detailed log #11 preferences.js:184:15
preferences.js - v2 detailed log #12 preferences.js:188:13
```

Probably interesting: when entering the second password-form first, and also entering the first password-form second, it still fails. The detailed log of that session as follows:

```
background.js - v2 detailed log #1a background.js:3:11
WebExtensions: Waiting for master password. implementation.js:127
WebExtensions: Waiting for master password. implementation.js:127
WebExtensions: Waiting for master password. implementation.js:127
WebExtensions: Waiting for master password. 4 implementation.js:127
WebExtensions: Master password has been entered. implementation.js:130
background.js - v2 detailed log #1b background.js:7:11
background.js - v2 detailed log #2 background.js:23:11
preferences.js - v2 detailed log #1 preferences.js:95:13
preferences.js - v2 detailed log #2 preferences.js:99:13
preferences.js - v2 detailed log #3 preferences.js:109:13
preferences.js - v2 detailed log #4 preferences.js:114:13
preferences.js - v2 detailed log #5a preferences.js:120:15
preferences.js - v2 detailed log #5b preferences.js:134:19
Error: An unexpected error occurred preferences.js:155:19
TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
Uncaught (in promise) TypeError: self.sleep is not a function
    init moz-extension://b04a7078-6670-0c4f-be73-e9c4e0ea8f46/chrome/content/scripts/preferences.js:129
preferences.js:129:16
```

**Comment by SamuelPlentz** [comment]
Sorry I didn't test my sleep code. Fixed it here:
[quicktext-3.3a6-tb.zip](https://github.com/jobisoft/quicktext/files/5379640/quicktext-3.3a6-tb.zip)

I also tried to fix the version of jobisoft. Could you try if it works? What is shown on startup?
```
Main window state is: ...
Waiting for mail-startup-done. #1
Waiting for mail-startup-done. #2
Startup seems to be done.
```
[quicktext-3.3aj1.zip](https://github.com/jobisoft/quicktext/files/5379708/quicktext-3.3aj1.zip)

The fixed version from jobisoft should probably work regardless if you dismiss/enter first or second Masterpassword prompt.

**Comment by hagedose** [comment]
What do you know, with 3.3a6 QT seems to be working for me even when a master password is set. This is what gets logged:
```
WebExtensions: Waiting for master password. implementation.js:127
WebExtensions: Waiting for master password. implementation.js:127
WebExtensions: Waiting for master password. 5 implementation.js:127
WebExtensions: Master password has been entered. implementation.js:130
background.js - v2 detailed log #1b background.js:7:11
background.js - v2 detailed log #2 background.js:23:11
preferences.js - v2 detailed log #1 preferences.js:95:13
preferences.js - v2 detailed log #2 preferences.js:99:13
preferences.js - v2 detailed log #3 preferences.js:109:13
preferences.js - v2 detailed log #4 preferences.js:114:13
preferences.js - v2 detailed log #5a preferences.js:120:15
preferences.js - v2 detailed log #5b preferences.js:134:19
Error: An unexpected error occurred preferences.js:155:19
TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
preferences.js - v2 detailed log #5b preferences.js:134:19
Error: An unexpected error occurred preferences.js:155:19
TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
preferences.js - v2 detailed log #5b preferences.js:134:19
Error: An unexpected error occurred preferences.js:155:19
TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
preferences.js - v2 detailed log #5b preferences.js:134:19
preferences.js - v2 detailed log #5c preferences.js:140:19
preferences.js - v2 detailed log #5d preferences.js:151:19
preferences.js - v2 detailed log #7 preferences.js:159:15
preferences.js - v2 detailed log #8 preferences.js:167:15
preferences.js - v2 detailed log #12 preferences.js:188:13
background.js - v2 detailed log #3 background.js:27:11
background.js - v2 detailed log #4 background.js:49:11
background.js - v2 detailed log #5 background.js:66:11
background.js - v2 detailed log #6 background.js:70:11
background.js - v2 detailed log #7 background.js:74:11
background.js - v2 detailed log #8 background.js:80:11
background.js - v2 detailed log #9 background.js:86:11
background.js - v2 detailed log #10 background.js:90:11
preferences.js - v2 detailed log #1 preferences.js:95:13
preferences.js - v2 detailed log #2 preferences.js:99:13
preferences.js - v2 detailed log #3 preferences.js:109:13
preferences.js - v2 detailed log #4 preferences.js:114:13
preferences.js - v2 detailed log #9 preferences.js:171:15
preferences.js - v2 detailed log #10 preferences.js:176:15
preferences.js - v2 detailed log #11 preferences.js:184:15
preferences.js - v2 detailed log #12
```

3.3aj1 does not work:

```
WebExtensions: Main window state is: loading implementation.js:415
WebExtensions: Waiting for mail-startup-done. #2 implementation.js:416
TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
Uncaught (in promise) Error: An unexpected error occurred undefined
DKIM_Verifier.background	FATAL	 Initializing failed with: Error: An unexpected error occurred background.mjs.js:41:34
TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] 2 L10nRegistry.jsm:658:19
CopySent2Current: 3.1 on Thunderbird 78.3.2
1602705561897	addons.webextension.<unknown>	WARN	Loading extension 'null': Reading manifest: Warning processing legacy: An unexpected property was found in the WebExtension manifest.
1602705561906	addons.webextension.<unknown>	WARN	Loading extension 'null': Reading manifest: Warning processing legacy: An unexpected property was found in the WebExtension manifest.
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] 2 L10nRegistry.jsm:658:19
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] 2 L10nRegistry.jsm:658:19
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] 2 L10nRegistry.jsm:658:19
WebExtensions: mail-startup-done has been seen. implementation.js:408
WebExtensions: Main window state is: complete implementation.js:409
WebExtensions: Waiting for master password. implementation.js:129
WebExtensions: Waiting for master password. implementation.js:129
WebExtensions: Waiting for master password. 7 implementation.js:129
WebExtensions: Master password has been entered. implementation.js:132
TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
Uncaught (in promise) Error: An unexpected error occurred
```

For now I will stick with 3.3a6

**Comment by jobisoft** [comment]
@hagedose : Could you try this version for me?
[quicktext.zip](https://github.com/jobisoft/quicktext/files/5384210/quicktext.zip)

And a general question: Do you get the master password prompt before the main window opens, or after?

**Comment by hagedose** [comment]
Doesn't work:

```
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] 2 L10nRegistry.jsm:658:19
TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
Uncaught (in promise) Error: An unexpected error occurred undefined
TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] 2 L10nRegistry.jsm:658:19
CopySent2Current: 3.1 on Thunderbird 78.3.2
1602758970099	addons.webextension.<unknown>	WARN	Loading extension 'null': Reading manifest: Warning processing legacy: An unexpected property was found in the WebExtension manifest.
1602758970111	addons.webextension.<unknown>	WARN	Loading extension 'null': Reading manifest: Warning processing legacy: An unexpected property was found in the WebExtension manifest.
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] 2 L10nRegistry.jsm:658:19
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] 2 L10nRegistry.jsm:658:19
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] 2 L10nRegistry.jsm:658:19
WebExtensions: Waiting for master password. implementation.js:127
WebExtensions: Waiting for master password. 12 implementation.js:127
WebExtensions: Master password has been entered. implementation.js:130
TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
Uncaught (in promise) Error: An unexpected error occurred undefined
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] 2 L10nRegistry.jsm:658:19
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] 2 L10nRegistry.jsm:658:19
prefs is undefined cs2c_implementation.js:373
Error: An unexpected error occurred undefined
```

The first password prompt opens before the main window, but then the main window opens in the back, and a second password prompt opens of top of everything.

**Comment by jobisoft** [comment]
> TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26

@hagedose: Can you expand that error message and post the stacktrace?

**Comment by jobisoft** [comment]
@EmmyGraugans : Could you check, if the version posted in https://github.com/jobisoft/quicktext/issues/167#issuecomment-709144213 works for you? If not, do you also see an error related to "transactions" in the error console (CTRL+SHIFT+J)?

**Comment by EmmyGraugans** [comment]
will do, when I get home (one to a few hours from now, sorry)

**Comment by jobisoft** [comment]
@EmmyGraugans : Do not be sorry. We have to thank you for your testing time!

**Comment by hagedose** [comment]
> > TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
> 
> @hagedose: Can you expand that error message and post the stacktrace?

<img width="1445" alt="Bildschirmfoto 2020-10-15 um 14 15 38" src="https://user-images.githubusercontent.com/1461803/96121884-f62cc080-0ef0-11eb-8ac8-f6f568940b75.png">

**Comment by jobisoft** [comment]
@hagedose: I believe this bug is fixed in the next point release of TB:
https://archive.mozilla.org/pub/thunderbird/candidates/78.3.3-candidates/build1/

If you are brave, you could backup your profile folder and run the candidate:
* copy it, while TB is not running
* name the copy something like "BACKUP_NO_TOUCH"
* install the candidate (in a seperate directory) and see if it works or not.

If you do not want to stay on the candidate, uninstall it again and then restore the profile folder:
* rename the original folder by appending a "_candidate"
* copy your "BACKUP_NO_TOUCH" again and give the new copy it the original name
* keep the backup just in case

**Comment by hagedose** [comment]
Thanks, I can confirm that 78.3.3 fixes the problem for QT, and also for "Copy Sent To Current". Now there is no more a password prompt before the main window opens.

**Comment by jobisoft** [comment]
@hagedose: Thanks for your tests!

To recap: That means, my last version (which is a release candidate for the next version of QT) posted here: https://github.com/jobisoft/quicktext/issues/167#issuecomment-709144213 work for you in 78.3.3?

**Comment by hagedose** [comment]
Yes!

**Comment by EmmyGraugans** [comment]
Very quick answer, since Safari *really* didn't like me typing a long(ish) reply (I tried three times, and three times, Safari either quit or reloaded the page, losing the text): Yes, that combination works. Logs are identical to what @hagedose wrote, so I won't repost them.

Maybe I'll repost the longer answer later today, did some more tests, but got bitten writing them down. Note to German people who switch between macOS and Linux or Windows: Command+Q is NOT going to insert an @ sign...

**Comment by SamuelPlentz** [comment]
Just for the record. The [new version](https://github.com/jobisoft/quicktext/issues/167#issuecomment-709144213) has no errors (AFAIS) here in TB 78.3.2 and works fine. (not that I could reproduce the errors reported here). Just waiting for a [rainy weekend](https://github.com/jobisoft/quicktext/pull/183) now. ;)

Strange that the try catch in my last version did work too. That was unexpected, as there was no exception previosly. Puzzles me.

Would be interesting if the [new version](https://github.com/jobisoft/quicktext/issues/167#issuecomment-709144213) and [TB 78.3.3](https://github.com/jobisoft/quicktext/issues/167#issuecomment-709286969) works fine in the strange environment of @BayouBillLa.

**Comment by BayouBillLa** [comment]
I get a 404 Page Not Found error when clicking your "new version" link.

**Comment by SamuelPlentz** [comment]
> I get a 404 Page Not Found error when clicking your "new version" link.

My fault fixed the links. Its the last version jobisoft posted.

**Comment by BayouBillLa** [comment]
Okay, I've installed QT3.3 on TB 78.3.2 (32-bit). So far, so good. Will post back if I encounter any problems. Otherwise, no news is good news. Thanks!

**Comment by jobisoft** [comment]
Thunderbird 78.3.3 is officially released. My Quicktext 3.3 posted here before which seemed to work with TB 78.3.3 has some special code for waiting for the master password, which I think is no longer needed.

Here is Quicktext 3.4 without that. Could you all test it with TB 78.3.3 ? 
[quicktext.zip](https://github.com/jobisoft/quicktext/files/5394475/quicktext.zip)

Depending on the feedback, I will release 3.3 or 3.4.

**Comment by hagedose** [comment]
Yes, Quicktext 3.4 works fine with the official release of Thunderbird 78.3.3. Thanks!

**Comment by BayouBillLa** [comment]
I've now updated QT to 3.4 and TB to 78.3.3 (32-bit), and all is well.

**Comment by EmmyGraugans** [comment]
Weelllll... 
The QuickText 3.4 does work, if I enter the master-password *the first time it appears*. The QuickText 3.3 with "waiting" code does work, if I enter the master-password *at any time*... (Even after cancelling a few times without entering a password.)

Log from QT 3.4:
```
TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
Uncaught (in promise) Error: An unexpected error occurred undefined
NS_ERROR_ABORT: User canceled master password entry 3 crypto-SDR.js:195
NS_ERROR_ABORT: User canceled master password entry 3 crypto-SDR.js:195
NS_ERROR_ABORT: User canceled master password entry 4 crypto-SDR.js:195
runnablePrompter:run: [Exception... "User canceled master password entry"  nsresult: "0x80004004 (NS_ERROR_ABORT)"  location: "JS frame :: resource://gre/modules/crypto-SDR.js :: decrypt :: line 195"  data: no]
MsgAsyncPrompter.jsm:50
NS_ERROR_ABORT: User canceled master password entry 3 crypto-SDR.js:195
```

Detail of "TransactionInactiveError":
```
TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished. IndexedDB.jsm:160:26
    method resource://gre/modules/IndexedDB.jsm:160
    get resource://gre/modules/ExtensionStorageIDB.jsm:309
    InterpretGeneratorResume self-hosted:1151
    AsyncFunctionNext self-hosted:693
    openWindowPrompt resource://gre/modules/Prompter.jsm:1229
    openPrompt resource://gre/modules/Prompter.jsm:1087
    openPromptSync resource://gre/modules/Prompter.jsm:1070
    nsIPrompt_promptPassword resource://gre/modules/Prompter.jsm:1559
    promptPassword resource://gre/modules/Prompter.jsm:1291
    decrypt resource://gre/modules/crypto-SDR.js:181
    _decryptLogins resource://gre/modules/storage-json.js:779
    findLogins resource://gre/modules/storage-json.js:638
    findLogins resource://gre/modules/LoginManager.jsm:488
    _promiseAuthPrompt resource:///modules/MsgAsyncPrompter.jsm:25
    _promiseAuthPrompt resource:///modules/MsgAsyncPrompter.jsm:23
    run resource:///modules/MsgAsyncPrompter.jsm:48
    InterpretGeneratorResume self-hosted:1151
    AsyncFunctionNext self-hosted:693
```

Log from QT3.3:
```
WebExtensions: Waiting for master password. implementation.js:127
WebExtensions: Waiting for master password. implementation.js:127
NS_ERROR_ABORT: User canceled master password entry 3 crypto-SDR.js:195
WebExtensions: Waiting for master password. implementation.js:127
NS_ERROR_ABORT: User canceled master password entry 3 crypto-SDR.js:195
WebExtensions: Waiting for master password. implementation.js:127
NS_ERROR_ABORT: User canceled master password entry 4 crypto-SDR.js:195
runnablePrompter:run: [Exception... "User canceled master password entry"  nsresult: "0x80004004 (NS_ERROR_ABORT)"  location: "JS frame :: resource://gre/modules/crypto-SDR.js :: decrypt :: line 195"  data: no]
MsgAsyncPrompter.jsm:50
WebExtensions: Waiting for master password. 6 implementation.js:127
WebExtensions: Waiting for master password. implementation.js:127
NS_ERROR_ABORT: User canceled master password entry 3 crypto-SDR.js:195
WebExtensions: Waiting for master password. implementation.js:127
WebExtensions: Waiting for master password. implementation.js:127
NS_ERROR_ABORT: User canceled master password entry 3 crypto-SDR.js:195
WebExtensions: Waiting for master password. 2 implementation.js:127
WebExtensions: Waiting for master password. 3 implementation.js:127
WebExtensions: Master password has been entered. implementation.js:130
```

Looks to me that some sort of "waiting" would be useful, if somebody enters for whichever reason the master-password not right away, but only a little bit later.

**Comment by jobisoft** [comment]
@EmmyGraugans : Did you do this test with TB 78.3.3 (not 78.3.2) which is shipped via auto update for 2 days?

**Comment by EmmyGraugans** [comment]
Yupp - 78.3.3 updated today (not the beta version from a few days ago)

**Comment by SamuelPlentz** [comment]
It looks like it would be better to use the 3.3 version then.
IMO we should publish this version soon.

**Comment by jobisoft** [comment]
Pending for review (but I cannot review my own add-ons).

**Comment by jobisoft** [comment]
3.3 is released, but I still get bad reports for this version, even on windows 10.

**Comment by SamuelPlentz** [comment]
As a [comment writer](https://addons.thunderbird.net/de/thunderbird/addon/quicktext/reviews/1164001/) states that it worked before in version 3.2 I think the issue with the new reporters is not the same as the issues in this thread. (and because everyone was happy with the latest and published version here)

**Comment by Clicac** [comment]
Using 3.3 on Win 10: Quicktext alway is disabled, can enable it with switching QT-AddOn off and on.

**Comment by SamuelPlentz** [comment]
@Clicac:
Can you try to disable all other addons? Does it stay active after restart?

**Comment by Clicac** [comment]
> @Clicac:
> Can you try to disable all other addons? Does it stay active after restart?

Yes, this seemingly helps: Disabling all AddOns: Quicktext works at once after startup of TB.
After some trials it seems, that the CardBook-AddOn is causing (?) the problem: Quicktext is always active after disabling CardBook (V53.4)... Hope that helps.

**Comment by jobisoft** [comment]
Can you all try this one?
[quicktext_3.4pre3.zip](https://github.com/jobisoft/quicktext/files/5489117/quicktext_3.4pre3.zip)

This may be a promising candidate.

**Comment by jobisoft** [comment]
Closing this as duplicate of #193
